### PR TITLE
Support interpreter re-entry for CRIU debug support

### PR DESCRIPTION
**Pull request by gacholio**
Add a new private flag which instructs the interpreter to exit and re-invoke itself. This will be used by CRIU when a restored image requests debug capabilities (by changing the interpreter entry point to the debug interpreter).

Related: #17642

### CODE REVIEW

**Comment by gacholio** [comment]
Use `VM_VMHelpers::requestInterpreterReentry(targetThread)` to request re-entry.

**Comment by tajila** [comment]
jenkins test sanity win jdk8

**Comment by tajila** [comment]
jenkins test sanity plinux jdk21

File: runtime/vm/arm64cinterp.m4 [hunk 1]
```diff
@@ -49,6 +49,8 @@ FUNC_LABEL(cInterpreter):
 	blr x28
 	cmp x0, {#}J9TR_bcloop_exit_interpreter
 	beq .L_cInterpExit
+	cmp x0, {#}J9TR_bcloop_reenter_interpreter
+	beq FUNC_LABEL(cInterpreter
```
Comment (keithc-ca) [inline runtime/vm/arm64cinterp.m4:53]: Missing closing parenthesis?

**Pull request merged**